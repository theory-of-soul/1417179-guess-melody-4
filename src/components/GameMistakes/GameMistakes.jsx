import React from "react";
import PropTypes from "prop-types";

const GameMistakes = ({amountMistakes}) => {
  const mistakes = new Array(amountMistakes).fill(null);

  return (
    <div className="game__mistakes">
      {mistakes.map((empty, i) => <div className="wrong" key={i}/>)}
    </div>
  );
};

GameMistakes.propTypes = {
  amountMistakes: PropTypes.number.isRequired
};

export default GameMistakes;
