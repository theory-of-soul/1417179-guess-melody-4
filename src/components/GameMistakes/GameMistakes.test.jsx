import React from "react";
import renderer from 'react-test-renderer';
import GameMistakes from "./GameMistakes";

describe(`GameMistakes snapshot tests`, () => {
  it(`GameMistakes render zero error`, () => {
    const component = renderer
      .create(<GameMistakes amountMistakes={0}/>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it(`GameMistakes render one error`, () => {
    const component = renderer
      .create(<GameMistakes amountMistakes={1}/>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it(`GameMistakes render two errors`, () => {
    const component = renderer
      .create(<GameMistakes amountMistakes={2}/>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it(`GameMistakes render three errors`, () => {
    const component = renderer
      .create(<GameMistakes amountMistakes={3}/>)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
