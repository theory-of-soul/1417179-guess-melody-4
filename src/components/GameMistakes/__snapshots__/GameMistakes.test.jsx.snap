// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameMistakes snapshot tests GameMistakes render one error 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="wrong"
  />
</div>
`;

exports[`GameMistakes snapshot tests GameMistakes render three errors 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
</div>
`;

exports[`GameMistakes snapshot tests GameMistakes render two errors 1`] = `
<div
  className="game__mistakes"
>
  <div
    className="wrong"
  />
  <div
    className="wrong"
  />
</div>
`;

exports[`GameMistakes snapshot tests GameMistakes render zero error 1`] = `
<div
  className="game__mistakes"
/>
`;
